### PR TITLE
Netty upgrade to 4.1.110 in grpc-java

### PR DESCRIPTION
**Pull request by kannanjgithub**
Netty upgrade to 4.1.110 in grpc-java
Issue #11150

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -399,6 +399,7 @@ grpc-netty version | netty-handler version | netty-tcnative-boringssl-static ver
 1.57.x-1.58.x      | 4.1.93.Final          | 2.0.61.Final
 1.59.x             | 4.1.97.Final          | 2.0.61.Final
 1.60.x-            | 4.1.100.Final         | 2.0.61.Final
+1.61.x-            | 4.1.110.Final         | 2.0.65.Final
```
Comment (ejona86) [inline SECURITY.md:402]: The previous line should have 1.60.x-1.65.x and this line should have 1.66.x-

**Review by ejona86 — COMMENTED** [review]
`MODULE.bazel` also needs updating. That is why the Bazel test is failing.

File: SECURITY.md (hunk 1 above)
Comment (kannanjgithub) [reply SECURITY.md]: Done.

**Comment by kannanjgithub** [comment]
> `MODULE.bazel` also needs updating. That is why the Bazel test is failing.

Done.

**Comment by XN137** [comment]
justFYI: `4.1.111.Final` would be out https://netty.io/news/2024/06/11/4-1-111-Final.html

**Comment by lhotari** [comment]
> justFYI: `4.1.111.Final` would be out https://netty.io/news/2024/06/11/4-1-111-Final.html

I reported #11284 about 4.1.111.Final. It would be useful to run CI with it to see what breaks.

**Review by ejona86 — APPROVED** [review]

**Comment by kannanjgithub** [comment]
@ejona86 Windows test is timing out. Any idea about this [test](https://github.com/grpc/grpc-java/blob/master/servlet/src/tomcatTest/java/io/grpc/servlet/TomcatInteropTest.java)?

**Comment by ejona86** [comment]
It can flake. Follow the link to Test Fusion and click the "rebuild" button.

**Comment by ejona86** [comment]
This needs to be backported to the 1.66 branch. Easiest way is to use https://gist.github.com/sergiitk/af606cbab72b00d8db8a53d21e766ee9 (`./backport.sh grpc-java 11273 66`). It does some niceties like adding "(v1.66.x backport)" to the PR description, adding a reference back to the original PR so you can see the list of all backports in one location. But it is also essentially just a cherry-pick onto a branch, so isn't mandatory.

**Comment by ejona86** [comment]
Reverting in #11440

**Comment by kannanjgithub** [comment]
I had thought once certain changes are committed to a branch https://screenshot.googleplex.com/86pkpKBjbbagzvR they belong to that branch. But what is happening is that they get carried over when I switch branch and get committed again with a different hash. I don't really understand how this works.

**Comment by ejona86** [comment]
Commits stay on branches, but certain operations can copy commits between branches, like rebase and merge. I could probably look at your bash history to figure out how it happened. It does seem like it is happening very frequently for you.

**Pull request merged**